### PR TITLE
jq: 1.6 -> 2022-05-26

### PR DESCRIPTION
**Pull request by edef1c**
###### Description of changes

jq has not seen a release in four years, despite a bug that silently mangles large numbers being fixed several years ago. Let's just use git HEAD instead.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by lf- — APPROVED** [review]

**Comment by lf-** [comment]
merge conflict fyi

**Review by lf- — APPROVED** [review]
Nice. `fetchTarball` is a good solution to avoiding depending on GitHub archive hashes

**Review by cole-h — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/jq/default.nix [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
 stdenv.mkDerivation rec {
   pname = "jq";
-  version = "1.6";
+  version = "2022-05-26";
 
   # Note: do not use fetchpatch or fetchFromGitHub to keep this package available in __bootPackages
-  src = fetchurl {
-    url = "https://github.com/stedolan/jq/releases/download/jq-${version}/jq-${version}.tar.gz";
-    sha256 = "sha256-XejI4pqqP7nMa0e7JymfJxNU67clFOOsytx9OLW7qnI=";
+  src = builtins.fetchTarball {
```
[the remark this answers is not in the document]
Comment (lf-) [reply pkgs/development/tools/jq/default.nix:14]: The trouble is that the hash is then of the archive itself which is not guaranteed stable. that said, bazel seems to have Hyrum's law'd it onto github's priority list to keep them stable even though it's not guaranteed.

[the remark this answers is not in the document]
Comment (cole-h) [reply pkgs/development/tools/jq/default.nix:14]: Ah, sorry -- I meant `fetchzip`, which will unpack the resulting tarball and gives me the same hash:

```
$ nix repl '<nixpkgs>'
# [...snip...]
nix-repl> :b pkgs.fetchzip { url = "https://github.com/stedolan/jq/archive/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8.tar.gz"; sha256 = ""; }
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
error: hash mismatch in fixed-output derivation '/nix/store/hhilyllp37xpf5shzmnjf1bdkr3fis3d-source.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-gsngye7EAmdSqH7ZH1g8/sQFj1nJWEGHpt48T8qXjZ0=

nix-repl> builtins.fetchTarball { url = "https://github.com/stedolan/jq/archive/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8.tar.gz"; sha256 = ""; }
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
error:
       … while calling the 'fetchTarball' builtin

         at «string»:1:1:

            1| builtins.fetchTarball { url = "https://github.com/stedolan/jq/archive/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8.tar.gz"; sha256 = ""; }
             | ^

       error: hash mismatch in file downloaded from 'https://github.com/stedolan/jq/archive/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8.tar.gz':
         specified: sha256:0000000000000000000000000000000000000000000000000000
         got:       sha256:17cdjz54yg6yls3l2n69b67hbi7y7ic1znbym196f0n4xv4y1jc2

nix-repl> 
$ nix hash to-sri 'sha256:17cdjz54yg6yls3l2n69b67hbi7y7ic1znbym196f0n4xv4y1jc2'
sha256-gsngye7EAmdSqH7ZH1g8/sQFj1nJWEGHpt48T8qXjZ0=
```

`builtins.fetchTarball` is not allowed in Nixpkgs (and is why ofborg fails): it moves the source fetching from build time to eval time, which can easily balloon said eval time.

[the remark this answers is not in the document]
Comment (lf-) [reply pkgs/development/tools/jq/default.nix:14]: Right. fetchFromGitHub is a thin wrapper around fetchzip which i imagine isn't going to be acceptable per the comment above, however i may be wrong.

[the remark this answers is not in the document]
Comment (cole-h) [reply pkgs/development/tools/jq/default.nix:14]: Oh, indeed, you are right. I missed that comment.

[the remark this answers is not in the document]
Comment (nikamarisa) [reply pkgs/development/tools/jq/default.nix:14]: How about manually doing what `fetchzip` does? It avoids depending on `unzip` but should avoid the issue um unstable archive hashes.

```suggestion
  src = fetchurl {
    url = "https://github.com/stedolan/jq/archive/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8.tar.gz";
    sha256 = "sha256-gsngye7EAmdSqH7ZH1g8/sQFj1nJWEGHpt48T8qXjZ0=";

    recursiveHash = true;
    downloadToTemp = true;

    postFetch = ''
      unpackDir="$TMPDIR/unpack"
      mkdir "$unpackDir"
      cd "$unpackDir"
      renamed="$TMPDIR/jq.tar.gz"
      mv "$downloadedFile" "$renamed"
      tar -xf "$renamed"
      chmod -R +w "$unpackDir"
      fn=$(cd "$unpackDir" && ls -A)
      mv "$unpackDir/$fn" "$out"
      chmod 755 "$out"
    '';
  };
```

[the remark this answers is not in the document]
Comment (ajs124) [reply pkgs/development/tools/jq/default.nix:14]: I'm not quite following, what's the issue with just using `fetchFromGitHub`?

[the remark this answers is not in the document]
Comment (ajs124) [reply pkgs/development/tools/jq/default.nix:14]: Ah, so jq is used during bootstrapping? That's the piece of information I was missing.

**Comment by lheckemann** [comment]
Looking at https://github.com/stedolan/jq/issues/2305, https://github.com/jqlang/jq might be sensible to use as the "new upstream". It seems to be run by the (most recently) active maintainers of stedolan/jq.

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/246515

**Pull request closed without merging**